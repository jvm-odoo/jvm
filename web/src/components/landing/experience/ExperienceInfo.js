import React, { Fragment } from 'react'

import { Text } from '../../common'
import { ExperiencePlace } from '../'

export const ExperienceInfo = ({ place, from, to }) => {
  return (
    <div>
      <Text small>
        <ExperiencePlace>{place}</ExperiencePlace>
      </Text>
      <Text small bold color='#E1E1E1'>&nbsp;/&nbsp;</Text>
      <Text small bold color='#6057F9'>
        From {from}
      </Text>
      {to && (
        <Fragment>
          <Text small bold color='#E1E1E1'>&nbsp;-&nbsp;</Text>
          <Text small bold color='#ABC5D4'>{to}</Text>
        </Fragment>
      )}
    </div>
  )
}