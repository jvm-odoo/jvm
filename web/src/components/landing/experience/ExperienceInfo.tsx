import React, { Fragment } from 'react'
import styled from 'styled-components'

import { Text } from '../../common'
import { ExperiencePlace } from '../'
import { IExperience } from './Experience'

const ExperienceInfoContainer = styled.div`
    display: flex;
    flex-direction: column;
    @media only screen and (min-width: 768px) {
        flex-direction: row;
    }
`

const DatesContainer = styled.div`
    display: flex;
`

export const ExperienceInfo = ({
    place,
    dateFrom,
    dateTo,
}: Partial<IExperience>) => {
    return (
        <ExperienceInfoContainer>
            <Text small>
                <ExperiencePlace>{place}</ExperiencePlace>
            </Text>
            <Text small bold color="#E1E1E1" mobileInvisible>
                &nbsp;/&nbsp;
            </Text>
            <DatesContainer>
                <Text small bold color="#6057F9">
                    From {dateFrom}
                </Text>
                {dateTo && (
                    <Fragment>
                        <Text small bold color="#E1E1E1">
                            &nbsp;-&nbsp;
                        </Text>
                        <Text small bold color="#597A82">
                            {dateTo}
                        </Text>
                    </Fragment>
                )}
            </DatesContainer>
        </ExperienceInfoContainer>
    )
}
