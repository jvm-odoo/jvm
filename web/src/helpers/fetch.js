export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
