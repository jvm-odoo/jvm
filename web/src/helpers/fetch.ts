export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}