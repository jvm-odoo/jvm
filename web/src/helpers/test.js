import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
