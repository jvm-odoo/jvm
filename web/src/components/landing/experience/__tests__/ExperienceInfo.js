import React from 'react'

import { ExperienceInfo } from '../ExperienceInfo'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<ExperienceInfo />))
