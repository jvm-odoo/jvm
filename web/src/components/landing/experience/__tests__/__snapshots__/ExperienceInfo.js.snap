// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.p
    small={true}
  >
    <styled.span />
  </styled.p>
  <styled.p
    bold={true}
    color="#E1E1E1"
    mobileInvisible={true}
    small={true}
  >
     / 
  </styled.p>
  <styled.div>
    <styled.p
      bold={true}
      color="#6057F9"
      small={true}
    >
      From 
    </styled.p>
  </styled.div>
</styled.div>
`;
